formats/AtomFormat.php

```diff
@@ -27,11 +36,33 @@ public function stringify(){
 
 		$entries = '';
 		foreach($this->getItems() as $item) {
-			$entryAuthor = $this->xml_encode($item->getAuthor());
-			$entryTitle = $this->xml_encode($item->getTitle());
-			$entryUri = $this->xml_encode($item->getURI());
-			$entryTimestamp = $this->xml_encode(date(DATE_ATOM, $item->getTimestamp()));
-			$entryContent = $this->xml_encode($this->sanitizeHtml($item->getContent()));
+			$entryTimestamp	= $item->getTimestamp();
+			$entryTitle		= $this->xml_encode($item->getTitle());
+			$entryContent	= $item->getContent();
+			$entryUri		= $item->getURI();
+
+			// the item id must be a valid unique URI
+			$entryID = $this->xml_encode($entryUri);
+			if (empty($entryID))
+				$entryID = 'urn:sha1:' . hash('sha1', $entryTitle . $entryContent);
+
+			if (empty($entryTimestamp))
+				$entryTimestamp = $this->lastModified;
+
+			if (empty($entryTitle)) {
+				$entryTitle = strip_tags($entryContent);
+				if (strlen($entryTitle) > self::LIMIT_TITLE)
+					$entryTitle = substr($entryTitle, 0, self::LIMIT_TITLE) . '...';
```

Comment:
The title should be kept as is. There is no limit in the specification and any generalized solution like this will fail in one case or another. Bridges are responsible to limit the title length themselves.

Reply:
Sorry, I didn't see the `if (empty($entryTitle))` part. :blush: 

One small suggestion to prevent splitting strings mid-word:

```php
$entryTitle = substr($entryTitle, 0, strpos(wordwrap($entryTitle, self::LIMIT_TITLE), "\n")) . '...';
```

It doesn't look pretty, but it splits the content into separate lines where each line is maximum `LIMIT_TITLE` long. Splits only take place on whitespace, so it prevents splitting mid-word effectively.

There is but one problem: If the input string already contains newlines, it will return everything up to the first newline if it's shorter than `LIMIT_TITLE`. Maybe use `str_replace("\n", ' ', $entryContent);` before `strip_tags` on line 53 to account for that.

Here is an example of the difference (using a title length of 40!)

```
$title = "This is a very long title explaining very important things\nNext sentence";
$title = substr($title, 0, strpos(wordwrap($title, 40), "\n")) . '...';
echo $title; // Returns "This is a very long title explaining..."
```

The current implementation would return a broken string.

```text
This is a very long title explaining ver...
```

Reply:
That's a great suggestion, thanks :)